src/main/java/com/mixpanel/android/viewcrawler/ViewCrawler.java

```diff
@@ -912,7 +938,38 @@ private void updateEditState() {
 
             mEditState.setEdits(editMap);
             mSeenExperiments.addAll(toTrack);
-            mVariantTracker.trackVariants(toTrack);
+
+            if (toTrack.size() > 0) {
```

Comment:
Why do you move this logic out from VariantTweak.trackVariants()? Do you think this is a better practice?